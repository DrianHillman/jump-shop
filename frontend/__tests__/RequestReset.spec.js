import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import { MockedProvider } from 'react-apollo/test-utils';
import RequestReset, { REQUEST_RESET_MUTATION } from '../components/RequestReset';

const mocks = [
  {
    request: {
      query: REQUEST_RESET_MUTATION,
      variables: {
        email: 'email@email.com',
      },
    },
    result: {
      data: { requestReset: { message: 'success', __typename: 'Message' } },
    },
  },
];

describe('<RequestReset />', () => {
  it('should render properly on the page and matches snapshot', async () => {
    const wrapper = mount(
      <MockedProvider>
        <RequestReset />
      </MockedProvider>
    );
    const form = wrapper.find('form[data-test="form"]');
    expect(toJSON(form)).toMatchSnapshot();
  });

  it('should call the mutation and respond to state', async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <RequestReset />
      </MockedProvider>
    );
    // simulate typing an email
    wrapper
      .find('input')
      .simulate('change', { target: { name: 'email', value: 'email@email.com' } });
    // submit the form
    wrapper.find('form').simulate('submit');
    await wait();
    wrapper.update();
    expect(wrapper.find('p').text()).toContain('Success! Check your email for a reset link!');
  });
});
