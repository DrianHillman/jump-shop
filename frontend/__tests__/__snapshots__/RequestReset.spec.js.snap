// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset /> should render properly on the page and matches snapshot 1`] = `
<form
  className="Form-sc-1t628lw-0 btQIbB"
  data-test="form"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Request a Password Reset
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label
      htmlFor="email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="email"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Request Reset!
    </button>
  </fieldset>
</form>
`;
